cmd2/parsing.py

```diff
@@ -0,0 +1,387 @@
+        result.output = output
+        result.outputTo = outputTo
+        result.pipeTo = pipeTo
+        result.suffix = suffix
+        result.multilineCommand = multilineCommand
+        return result
+
+    def parse_command_only(self, rawinput: str) -> Statement:
+        """Partially parse input into a Statement object. The command is
+        identified, and shortcuts and aliases are expanded.
+        Terminators, multiline commands, and output redirection are not
+        parsed.
+        """
+        # strip C-style comments
+        # shlex will handle the python/shell style comments for us
+        # save rawinput for later
+        rawinput = re.sub(self.comment_pattern, _comment_replacer, rawinput)
+        # we are going to modify line, so create a copy of the raw input
+        line = rawinput
+        command = None
+        args = ''
+
+        # expand shortcuts, have to do this first because
+        # a shortcut can expand into multiple tokens, ie '!ls' becomes
+        # 'shell ls'
+        line = self.expand_shortcuts(line)
+
+        # split the input on whitespace
+        s = shlex.shlex(line, posix=False)
+        s.whitespace_split = True
+        tokens = self.split_on_punctuation(list(s))
+
+        # expand aliases
+        if tokens:
+            command_to_expand = tokens[0]
+            tokens[0] = self.expand_aliases(command_to_expand)
+
+        (command, args) = self._command_and_args(tokens)
+
+        # build Statement object
+        result = Statement(args)
+        result.raw = rawinput
+        result.command = command
+        result.args = args
+        return result
+
+    def expand_shortcuts(self, line: str) -> str:
+        """Expand shortcuts at the beginning of input."""
+        for (shortcut, expansion) in self.shortcuts:
+            if  line.startswith(shortcut):
+                # If the next character after the shortcut isn't a space, then insert one
+                shortcut_len = len(shortcut)
+                if len(line) == shortcut_len or line[shortcut_len] != ' ':
+                    expansion += ' '
+
+                # Expand the shortcut
+                line = line.replace(shortcut, expansion, 1)
+                break
+        return line
+
+    def expand_aliases(self, command: str) -> str:
+        """Given a command, expand any aliases for the command"""
+        # make a copy of aliases so we can edit it
+        tmp_aliases = list(self.aliases.keys())
+        keep_expanding = len(tmp_aliases) > 0
+
+        while keep_expanding:
+            for cur_alias in tmp_aliases:
+                keep_expanding = False
+                if command == cur_alias:
+                    command = self.aliases[cur_alias]
+                    tmp_aliases.remove(cur_alias)
+                    keep_expanding = len(tmp_aliases) > 0
+                    break
+        return command
+
+    def _command_and_args(self, tokens: List[str]) -> Tuple[str, str]:
```

Comment:
Recommend making this a static method since ``self`` isn't used within

Reply:
Fixed and pushed to `ply` branch.